src/CheckIfDead.php

```diff
@@ -378,15 +377,52 @@ protected function sanitizeURL( $url ) {
 			);
 		}
 		if ( isset( $parts['query'] ) ) {
-			$url .= "?".urlencode( urldecode( $parts['query'] ) );
+			$url .= "?";
+			$parts['query'] = explode( '&', $parts['query'] );
+			foreach ( $parts['query'] as $i=>$argument ) {
+				$parts['query'][$i] = implode( '=',
+					array_map( "urlencode",
+						explode( '=', $parts['query'][$i], 2 )
+					)
+				);
+			}
+			$parts['query'] = implode( '&', $parts['query'] );
+			$url .= $parts['query'];
 		}
 		if ( isset( $parts['fragment'] ) ) {
-			$url .= "#".urlencode( urldecode( $parts['fragment'] ) );
+			$url .= "#".$parts['fragment'];
 		}
 		return $url;
 	}
 
 	/**
+	 * Custom parse_url function to support UTF-8 URLs
+	 *
+	 * @param string $url The URL to parse. Invalid characters are replaced by _.
+	 * @param int $component Only return a given component
+	 * @return mixed False on failure, array on success, string or null if component is specified.
+	 */
+	private function parse_url( $url, $component = -1 ) {
```

Comment:
Why is this function private? It would be better to rename this function to something else to avoid confusion with the inbuilt php function. Maybe something like <code>parse_url_utf8()</code>.